apps/examples/testcase/le_tc/network/tc_net_igmp.c

```diff
@@ -0,0 +1,363 @@
+* @brief				:
+* @scenario				:
+* @apicovered			: netif_set_default
+* @precondition			:
+* @postcondition		:
+*/
+static void default_netif_add(void)
+{
+	IP4_ADDR(&test_gw, 192, 168, 0, 1);
+	IP4_ADDR(&test_ipaddr, 192, 168, 0, 1);
+	IP4_ADDR(&test_netmask, 255, 255, 0, 0);
+	netif_set_default(netif_add(&test_netif, &test_ipaddr, &test_netmask, &test_gw, NULL, default_netif_init, NULL));
+	netif_set_up(&test_netif);
+}
+
+/**
+* @testcase				:
+* @brief				:
+* @scenario				:
+* @apicovered			: netif_remove
+* @precondition			:
+* @postcondition		:
+*/
+static void default_netif_remove(void)
+{
+	netif_remove(&test_netif);
+}
+
+/**
+* @testcase				: etharp_setup
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_remove_all
+* @precondition			:
+* @postcondition		:
+*/
+static void etharp_setup(void)
+{
+	etharp_remove_all();
+	default_netif_add();
+}
+
+/**
+* @testcase				: etharp_teardown
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_remove_all
+* @precondition			:
+* @postcondition		:
+*/
+static void etharp_teardown(void)
+{
+	etharp_remove_all();
+	default_netif_remove();
+}
+
+
+/**
+* @testcase				: tc_etharp_free_entry
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_free_entry
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_etharp_free_entry(void)
+{
+	int i =1;
+
+	etharp_free_entry(i);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+* @testcasu				: tc_free_etharp_q
+* @brief				:
+* @scenario				:
+* @apicovered			: free_etharp_q
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_free_etharp_q(void)
+{
+	struct etharp_q_entry *q;
+	q = (struct etharp_q_entry *) malloc (sizeof (struct etharp_q_entry));
+
+	free_etharp_q(q);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcasu				: tc_etharp_add_static_n
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_add_static
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_etharp_find_addr_n(void)
+{
+	int ret = 0;
+	etharp_find_addr(NULL, NULL, NULL, NULL);
+	TC_ASSERT_EQ("etharp_find_addr", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase				: tc_etharp_add_static_p
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_find_addr
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_etharp_find_addr_p(void)
+{
+	ip_addr_t *unused_ipaddr;
+	struct eth_addr *unused_ethaddr;
+	int ret = -1;
+	ip_addr_t adrs[ARP_TABLE_SIZE + 2];
+	int i;
+	for (i = 0; i < ARP_TABLE_SIZE + 2; i++) {
+		IP4_ADDR(&adrs[i], 192, 168, 0, i + 2);
+	}
+
+	ret = etharp_add_static_entry(&adrs[ARP_TABLE_SIZE], &test_ethaddr); 
+	etharp_find_addr(NULL, &adrs[ARP_TABLE_SIZE], &unused_ethaddr, &unused_ipaddr);
+	TC_ASSERT_NEQ("etharp_find_addr", ret, 1);
+	TC_SUCCESS_RESULT();
+}
+
+
+
+/**
+* @testcase				: tc_igmp_removegroup_n
+* @brief				:
+* @scenario				:
+* @apicovered			: igmp_removegroup_n
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_igmp_removegroup_n(void)
+{
+	int ret = -1;
+
+	ret = igmp_remove_group(NULL);
```

Comment:
same. How do you call it?

Reply:
We have written all the test cases in le_tc . If we need to write a test case for igmp_remove_group( ) then we have to make it non-static. For this API to be covered we created one macro CONFIG_TC_IGMP, under this macro we make that API as non-static. which will be enabled only for UTC test cases . for normal approach that API will remain static.

Reply:
@viany  as I said before, you should not call lwip API on TC. you can use ioctl() with SIOCSIPMSFILTER instead of igmp_remove_group()

Reply:
@pillip8282 :  I will take care of those in next patch